internal/util/util.go

```diff
@@ -111,3 +111,45 @@ func inspectPlatformOperator(po platformv1alpha1.PlatformOperator) error {
 func buildPOFailureMessage(name, reason string) error {
 	return fmt.Errorf("encountered the failing %s platform operator with reason %q", name, reason)
 }
+
+// InspectBundleDeployment is responsible for inspecting an individual BD
+// resource, and verifying whether the referenced Bundle contents has been
+// successfully unpacked and persisted to the cluster.
+func InspectBundleDeployment(_ context.Context, conditions []metav1.Condition) *metav1.Condition {
+	unpacked := meta.FindStatusCondition(conditions, rukpakv1alpha1.TypeHasValidBundle)
+	if unpacked == nil {
+		return &metav1.Condition{
+			Type:    platformtypes.TypeApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  platformtypes.ReasonUnpackPending,
+			Message: "Waiting for the bundle to be unpacked",
+		}
+	}
+	if unpacked.Status != metav1.ConditionTrue {
+		return &metav1.Condition{
+			Type:    platformtypes.TypeApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  unpacked.Reason,
+			Message: unpacked.Message,
+		}
+	}
+
+	applied := meta.FindStatusCondition(conditions, rukpakv1alpha1.TypeInstalled)
+	if applied == nil {
+		return &metav1.Condition{
+			Type:    platformtypes.TypeApplied,
+			Status:  metav1.ConditionFalse,
+			Reason:  platformtypes.ReasonUnpackPending,
+			Message: "Waiting for the bundle to be unpacked",
```

Comment:
Should this be more like "Waiting for bundle to be installed" with reason "InstallPending"?

Reply:
Nice catch. Updated in https://github.com/openshift/platform-operators/pull/41/commits/28b4316bc8c76cdb4d61f41877fbe6351ac50f39. I also renamed the "*Applied*" constant variables to "*Install*" as that felt more natural.